spec/ParseQuery.hint.spec.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const MongoStorageAdapter = require('../lib/Adapters/Storage/Mongo/MongoStorageAdapter')
+  .default;
+const mongoURI =
+  'mongodb://localhost:27017/parseServerMongoAdapterTestDatabase';
+const PostgresStorageAdapter = require('../lib/Adapters/Storage/Postgres/PostgresStorageAdapter')
+  .default;
+const postgresURI =
+  'postgres://localhost:5432/parse_server_postgres_adapter_test_database';
+const request = require('../lib/request');
+let databaseAdapter;
+
+const hintHelper = () => {
+  if (process.env.PARSE_SERVER_TEST_DB === 'postgres') {
+    if (!databaseAdapter) {
+      databaseAdapter = new PostgresStorageAdapter({ uri: postgresURI });
+    }
+  } else {
+    databaseAdapter = new MongoStorageAdapter({ uri: mongoURI });
+  }
+  const subjects = [
+    'coffee',
+    'Coffee Shopping',
+    'Baking a cake',
+    'baking',
+    'Café Con Leche',
+    'Сырники',
+    'coffee and cream',
+    'Cafe con Leche',
+  ];
+  const requests = [];
+  for (const i in subjects) {
+    const request = {
+      method: 'POST',
+      body: {
+        subject: subjects[i],
+        comment: subjects[i],
+      },
+      path: '/1/classes/TestObject',
+    };
+    requests.push(request);
+  }
+  return reconfigureServer({
+    appId: 'test',
+    restAPIKey: 'test',
+    publicServerURL: 'http://localhost:8378/1',
+    databaseAdapter,
+  })
+    .then(() => {
+      return request({
+        method: 'POST',
+        url: 'http://localhost:8378/1/schemas/TestObject',
+        body: {
+          className: 'TestObject',
+          fields: {
+            subject: {
+              type: 'String',
+            },
+            comment: {
+              type: 'String',
+            },
+          },
+          indexes: {
+            indexName: {
+              subject: 1,
+            },
+          },
+        },
+      });
+    })
+    .then(() => {
+      return request({
+        method: 'POST',
+        url: 'http://localhost:8378/1/batch',
+        body: {
+          requests,
+        },
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test',
+          'Content-Type': 'application/json',
+        },
+      });
+    });
+};
+
+describe('Parse.Query hint testing', () => {
+  it('should execute query with hint as a string', done => {
+    hintHelper()
+      .then(() => {
+        return request({
+          method: 'POST',
+          url: 'http://localhost:8378/1/classes/TestObject',
+          body: { where: {}, _method: 'GET', hint: 'subject' },
+          headers: {
+            'X-Parse-Application-Id': 'test',
+            'X-Parse-REST-API-Key': 'test',
+            'Content-Type': 'application/json',
+          },
+        });
+      })
+      .then(
+        resp => {
+          expect(resp.data.results.length).toBe(3);
```

Comment:
Can you change the tests to prove that hint is actually being used.

Show results before hint and after hint is used. I'm sure there is an example online somewhere that you can find.

Also check travis, do your tests run locally?

Reply:
The only way I would know to do this is to use the [explain](https://docs.mongodb.com/manual/reference/operator/meta/explain/) command when executing the query. I guess I can go in and add this functionality too?

I cannot get the tests to pass locally even when I pull the master branch without any changes.

Reply:
Let me pull this down and try. Is thats ok?

Reply:
of course!

Reply:
I can't push to your branch `query-hint` can you add me?